src/Forms/Shared/ApiKeyPrompt.xaml.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright 2021 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at: http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+// language governing permissions and limitations under the License.
+
+using ArcGISRuntime.Samples.Shared.Managers;
+using System;
+using Xamarin.Essentials;
+using Xamarin.Forms;
+using Xamarin.Forms.Xaml;
+
+namespace ArcGISRuntime
+{
+    [XamlCompilation(XamlCompilationOptions.Compile)]
+    public partial class ApiKeyPrompt : ContentPage
+    {
+        public ApiKeyPrompt()
+        {
+            InitializeComponent();
+            Initialize();
+        }
+
+        private void Initialize()
+        {
+            CurrentKeyText.Text = Esri.ArcGISRuntime.ArcGISRuntimeEnvironment.ApiKey;
+            UpdateValidiyText();
+        }
+
+        private async void UpdateValidiyText()
+        {
+            ApiKeyStatus status = await ApiKeyManager.CheckKeyValidity();
```

Comment:
Could this throw an exception? If so, it might not be handled?

Reply:
Its not possible for `CheckKeyValidity` to throw an exception. All paths return a value, and the only call that can throw an exception is in a try/catch.